File: 137_Single_Number_II/bit_operation.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int result = 0;
        for (int i = 0; i < 32; ++i) {
            int total = 0;
            for (auto & num : nums) {
                total += ((num >> i) & 1);
            }
            if (total % 3) {
                result |= (1 << i);
            }
        }
        return result;
    }
};

File: 137_Single_Number_II/bit_operation.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        result = 0
        for i in range(0, 32):
            total = 0
            for num in nums:
                total += ((num >> i) & 1)
            if total % 3:
                if i == 31:
                    result -= (1 << i)
                else:
                    result |= (1 << i)
        return result